packages/admin/dashboard/src/routes/locations/location-service-zone-shipping-option-pricing/location-service-zone-shipping-option-pricing.tsx

```diff
@@ -1,25 +1,17 @@
-import { json, useParams } from "react-router-dom"
+import { useParams } from "react-router-dom"
 
 import { RouteFocusModal } from "../../../components/modals"
-import { useShippingOptions } from "../../../hooks/api/shipping-options"
+import { useShippingOption } from "../../../hooks/api/shipping-options"
 import { EditShippingOptionsPricingForm } from "./components/create-shipping-options-form"
 
 export function LocationServiceZoneShippingOptionPricing() {
   const { so_id, location_id } = useParams()
 
-  const { shipping_options, isPending, isFetching, isError, error } =
-    useShippingOptions({
-      // TODO: change this when GET option by id endpoint is implemented
-      id: [so_id!],
+  const { shipping_option: shippingOption, isError, error } =
+    useShippingOption(so_id!, {
```

Comment:
Q: instead of forcing the so_id to be present should we check and narrow it instead to prevent bugs from raising ?